play.py

```diff
@@ -33,6 +34,27 @@ def do_GET(self):
         ## Think
         joystick = model.y.eval(feed_dict={model.x: [vec], model.keep_prob: 1.0})[0]
 
+
+        ### Pre-process the controls
+        #### There is a bias to drive straight so multiply x axis by a multiplier
+        x_mult = 2
+        joystick[0] = max(min(joystick[0] * x_mult, 1), -1)
+
+        #### Convert button pushes to 'probabilities' then sample 0/1
+        #### use some version of 1/(1 + np.exp(-joystick[n])) instead?
+        #### button 5 should be left trigger eventually (mod the plugin)
+        p2 = max(min(joystick[2], 1), 0)
+        joystick[2] = np.random.choice(a = [0,1], p = [1-p2, p2])
+
+        p3 = max(min(joystick[3], 1), 0)
+        joystick[3] = np.random.choice(a = [0,1], p = [1-p3, p3])
+
+        p4 = max(min(joystick[4], 1), 0)
+        joystick[4] = np.random.choice(a = [0,1], p = [1-p4, p4])
+
+        #p5 = max(min(joystick[5], 1), 0)
+        #joystick[5] = np.random.choice(a = [0,1], p = [1-p5, p5])
```

Comment:
can you remove the future work comments and create issues instead please.

Otherwise adding this multiplier seems fine to me. Its interesting that Nvidia found the same thing in their paper.

Can you move the convert_to_probability code to a utility function named the same? This will remove some duplicate code and make it easier to understand and change later. It would also be nice if we didn't have to add variables to this scope and could hide all of this in the function (aka remove p2, p3, p4)